examples/genomicsdb_common.py

```diff
@@ -0,0 +1,101 @@
+#
+# genomicsdb_common python script
+#
+# The MIT License
+#
+# Copyright (c) 2024 dātma, inc™
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import json
+import os
+import re
+
+import genomicsdb
+
+
+def normalize_path(path):
+    if "://" in path:
+        return path
+    else:
+        return os.path.abspath(path)
+
+
+def parse_vidmap_json(vidmap_file, intervals=None):
+    if isinstance(intervals, str):
+        is_file = True
+    else:
+        is_file = False
+    contigs_map = {}
+    vidmap = json.loads(genomicsdb.read_entire_file(vidmap_file))
+    contigs = vidmap["contigs"]
+    if intervals and is_file:
+        with open(intervals) as file:
+            intervals = [line.rstrip() for line in file]
+    all_intervals = not intervals
+    if not intervals:
+        intervals = []
+    for contig in contigs:
+        contigs_map[contig["name"]] = contig
+        if all_intervals:
+            intervals.append(contig["name"])
+    return contigs_map, intervals
+
+
+interval_pattern = re.compile(r"(.*):(.*)-(.*)|(.*):(.*)|(.*)")
+
+
+# get tiledb offsets for interval
+def parse_interval(interval: str):
```

Comment:
I dunno if this is better or worse than regex -- but maybe we could consider using string `split` (plus check for existence of the delimiter). Might be simpler

Reply:
This is done just while parsing intervals for the cli. I don't expect much time spent on this function. Guess using split is simpler.